corehq/apps/es/README.rst

```diff
@@ -80,6 +88,113 @@ This reindex procedure is inherently safe because:
   hoster are not outside the Dimagi main track.
 
 
+Design Details
+''''''''''''''
+
+Reindex Procedure Details
+'''''''''''''''''''''''''
+
+1. Configure multiplexing on an index.
+
+   - Configure the document adapter for the index with a "secondary index name".
+     This will cause the adapter to use multiplexing logic instead of a single
+     index.
+
+     **Note**: The multiplexing logic required for this operation is not yet
+     implemented. The multiplexing adapter will most likely delegate to two
+     document adapters configured for separate indexes. Suffice it to say that
+     when a secondary index is defined for an adapter, it effectively becomes a
+     multiplexing adapter (which to the consumer, is indistinguishable from a
+     "standard" adapter).
+
+   - *(Optional)* If the reindex involves other meta-index changes (shards,
+     mappings, etc), also update those configurations at this time.
+   - Add a migration which performs all cluster-level operations required for
+     the new (secondary) index. For example:
+
+     - creates the new index
+     - configures shards, replicas, etc for the index
```

Comment:
> Much of that configuration does not exist in HQ

Yes, these would remain mostly the same, the only difference being that these changes<sup>1</sup> will now be _applied_ by migrations.  Just like now, there can be "default" settings defined in code, which pull in override values from settings.  The migration is the trigger that will apply those values. The implementation might<sup>2</sup> need a "smart" migration utility that can skip application of configurations that are defined in settings in the event that the cluster is already in the desired state.

<sup>1</sup> "these changes" refers to _only_ the settings that require a reindex.  Tunables that do not require a reindex can be managed outside of migrations.

<sup>2</sup> It's possible that the Elasticsearch API for these settings are idempotent on the server-side, in which case it would be safe for the migration to _always_ apply them.